File: 953-reverse-only-letters/reverse-only-letters.cpp
class Solution {
public:
    string reverseOnlyLetters(string s) {
        vector<char> list;
        for(char ch:s){
            if(isalpha(ch)){
                list.push_back(ch);
            }
        }
        reverse(list.begin(),list.end());
        int j=0;
        for(int i=0;i<s.length();i++){
            if(isalpha(s[i])){
                s[i]=list[j];
                j++;
            }
        }
        return s;
    }
};

File: 953-reverse-only-letters/reverse-only-letters.java
class Solution {
    public String reverseOnlyLetters(String s) {
        List<Character> list=new ArrayList<>();
        char[] ch=s.toCharArray();
        for(char c:s.toCharArray()){
            if(Character.isLetter(c)){
                list.add(c);
            }
        }
        Collections.reverse(list);
        int j=0;
        for(int i=0;i<s.length();i++){
            if(Character.isLetter(ch[i])){
                ch[i]=list.get(j);
                j++;
            }
        }
        return new String(ch);
    }
}